app/src/org/commcare/android/view/GridEntityView.java

```diff
@@ -259,14 +266,19 @@ public void setViews(Context context, Detail detail, Entity entity){
 		// see if any entities have background data set
 		for(int i=0; i<bgData.length; i++){
 			if(!bgData[i].equals("")){
```

Comment:
It is good to re-write string comparisons like this so that the .equals is being called on the concrete value. This avoid null pointer exceptions.
For instance here, !"".equals(bgData[i])